src/Misc/PartitionWithCondition.jl

```diff
@@ -0,0 +1,103 @@
+mutable struct PartitionWithCondition
+  n::Int
+  k::Int
+  l::Int
+  sum::Int
+  weighted_sum::Int
+  vect::Vector{Int}
+  # The struct stores an integer vector, i.e. the vector [a_0, ..., a_{k-1}], together with
+  # sum = a_2 + ... + a_{k-1} and weighted_sum = 2*a_2 + 3*a_3 + ... + (k-1)*a_{k-1}.
+  # This yields faster computations.
+
+  function PartitionWithCondition(n::Int, k::Int, l::Int)
+    # Catch non-meaningful input
+    @req n>=0 "n should be >= 0"
+    @req k>0 "k should be > 0"
+    @req l>=0 "l should be >= 0"
+    return new(n, k, l, 0, 0, zeros(Int, k))
+  end
+
+end
+
+@doc raw"""
+  ``partition_with_condition(
+    n::Int,
+    k::Int,
+    l::Int
+  ) -> iterator``
+  
+  This code provides an iterator ``partition_with_condition(n, k, l)``. This
+  iterator enumerates all possible ``k``-tuples ``[ a_0, ..., a_{k-1} ]`` of
+  nonnegative integers satisfying ``a_0 + a_1 + ... + a_{k-1} = n`` and,
+  furthermore, ``0*a_0 + 1*a_1 + 2*a_2 + ... + (k-1)*a_{k-1} = l``.
```

Comment:
The following is more concise.

```suggestion
  Return an iterator over all possible ``k``-tuples ``[ a_0, ..., a_{k-1} ]`` of
  nonnegative integers satisfying ``a_0 + a_1 + ... + a_{k-1} = n`` and,
  furthermore, ``0*a_0 + 1*a_1 + 2*a_2 + ... + (k-1)*a_{k-1} = l``.
```